cmd/stellar-rpc/internal/methods/simulate_transaction.go

```diff
@@ -385,8 +385,16 @@ func getBucketListSizeAndProtocolVersion(
 	if !ok {
 		return 0, 0, fmt.Errorf("missing meta for latest ledger (%d)", latestLedger)
 	}
-	if closeMeta.V != 1 {
+	switch closeMeta.V {
+	case 1:
+		return uint64(closeMeta.V1.TotalByteSizeOfLiveSorobanState),
+			uint32(closeMeta.V1.LedgerHeader.Header.LedgerVersion),
+			nil
+	case 2:
```

Comment:
Should we also handle ingestion of `TransactionMetaV4` in this PR or separately?

Reply:
Uh, separately, I would think. Once we add integration tests for protocol 23

Reply:
Yep makes sense :+1: